iOS_SDK/OneSignalSDK/Source/OneSignalOutcomeEventsController.m

```diff
@@ -29,63 +29,71 @@ of this software and associated documentation files (the "Software"), to deal
 #import "OneSignalOutcomeEventsController.h"
 #import "Requests.h"
 #import "OSOutcomeEvent.h"
-#import "OSUniqueOutcomeNotification.h"
+#import "OSCachedUniqueOutcome.h"
 #import "OneSignalClient.h"
-#import "OneSignalSessionManager.h"
-#import "OSOutcomesUtils.h"
+#import "OSSessionManager.h"
+#import "OSOutcomeEventsRepository.h"
 #import "OneSignalUserDefaults.h"
 #import "OneSignalCommonDefines.h"
-#import "OSOutcomeEventsDefines.h"
+#import "OSInfluenceDataDefines.h"
+
+@interface OneSignalOutcomeEventsController ()
+
+@property (strong, nonatomic, readonly, nonnull) OSSessionManager *sessionManager;
+@property (strong, nonatomic, readonly, nonnull) OSOutcomeEventsFactory *outcomeEventsFactory;
+
+@end
 
 @implementation OneSignalOutcomeEventsController
 
 // Keeps track of unique outcome events sent for UNATTRIBUTED sessions on a per session level
 NSMutableSet *unattributedUniqueOutcomeEventsSentSet;
 
-// Keeps track of unique outcome events sent for ATTRIBUTED sessions on a per notification level
-NSMutableArray<OSUniqueOutcomeNotification *> *attributedUniqueOutcomeEventNotificationIdsSentArray;
-
-- (instancetype _Nonnull)init:(OneSignalSessionManager * _Nonnull)sessionManager {
+- (instancetype _Nonnull)initWithSessionManager:(OSSessionManager * _Nonnull)sessionManager
+                           outcomeEventsFactory:(OSOutcomeEventsFactory *)outcomeEventsFactory {
     if (self = [super init]) {
-        self.osSessionManager = sessionManager;
+        _sessionManager = sessionManager;
+        _outcomeEventsFactory = outcomeEventsFactory;
         [self initUniqueOutcomeEventsFromCache];
     }
     return self;
 }
 
 - (void)initUniqueOutcomeEventsFromCache {
-    NSSet *tempUnattributedUniqueOutcomeEventsSentSet = [self getUnattributedUniqueOutcomeEventNames];
+    NSSet *tempUnattributedUniqueOutcomeEventsSentSet = [_outcomeEventsFactory.repository getUnattributedUniqueOutcomeEventsSent];
     if (tempUnattributedUniqueOutcomeEventsSentSet)
         unattributedUniqueOutcomeEventsSentSet = [NSMutableSet setWithSet:tempUnattributedUniqueOutcomeEventsSentSet];
-
-    NSArray *tempAttributedUniqueOutcomeEventNotificationIdsSentArray = [self getAttributedUniqueOutcomeEventNotificationIds];
-    if (tempAttributedUniqueOutcomeEventNotificationIdsSentArray)
-        attributedUniqueOutcomeEventNotificationIdsSentArray = [NSMutableArray arrayWithArray:tempAttributedUniqueOutcomeEventNotificationIdsSentArray];
 }
 
 - (void)clearOutcomes {
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:@"Unique Outcome clear for current session"];
```

Comment:
Just rewording a bit, `Unique Outcome clear for current session`
`Outcomes cleared for current session`

We shouldn't assume this is only used for unique outcomes and if it is then we should change the name of the method instead of changing the log.